app/src/views/features/rules/RuleEditor/components/Header/ActionButtons/CreateRuleButton/index.jsx

```diff
@@ -155,79 +156,80 @@ const CreateRuleButton = ({
     }
   }, [isDisabled, MODE, premiumRuleLimitType, user.details?.isPremium]);
 
-  const handleOtherRuleEvents = useCallback(
-    async (disableTaskCompletedModal = false) => {
-      const otherRules = [RuleType.RESPONSE, RuleType.REDIRECT];
+  const handleOtherRuleEvents = useCallback(async () => {
+    const otherRules = [RuleType.RESPONSE, RuleType.REDIRECT];
 
-      if (otherRules.includes(currentlySelectedRuleData.ruleType)) {
-        const incentiveEvent =
-          currentlySelectedRuleData.ruleType === RuleType.RESPONSE
-            ? IncentivizeEvent.RESPONSE_RULE_CREATED
-            : IncentivizeEvent.REDIRECT_RULE_CREATED;
+    if (otherRules.includes(currentlySelectedRuleData.ruleType)) {
+      const incentiveEvent =
+        currentlySelectedRuleData.ruleType === RuleType.RESPONSE
+          ? IncentivizeEvent.RESPONSE_RULE_CREATED
+          : IncentivizeEvent.REDIRECT_RULE_CREATED;
 
-        claimIncentiveRewards({
-          type: incentiveEvent,
-          metadata: { rule_type: currentlySelectedRuleData.ruleType },
-        }).then((response) => {
-          if (response.data?.success) {
-            dispatch(incentivizationActions.setUserMilestoneDetails({ userMilestoneDetails: response.data?.data }));
+      claimIncentiveRewards({
+        dispatch,
+        isUserloggedIn: user?.loggedIn,
+        event: { type: incentiveEvent, metadata: { rule_type: currentlySelectedRuleData.ruleType } },
+      })?.then((response) => {
+        if (response.data?.success) {
+          dispatch(
+            incentivizationActions.setUserMilestoneAndRewardDetails({
+              userMilestoneAndRewardDetails: response.data?.data,
+            })
+          );
 
-            if (!disableTaskCompletedModal) {
-              dispatch(
-                actions.toggleActiveModal({
-                  modalName: "incentiveTaskCompletedModal",
-                  newValue: true,
-                  newProps: {
-                    event: incentiveEvent,
-                  },
-                })
-              );
-            }
-          }
-        });
-      }
-    },
-    [currentlySelectedRuleData.ruleType, dispatch]
-  );
+          dispatch(
+            incentivizationActions.toggleActiveModal({
+              modalName: IncentivizationModal.TASK_COMPLETED_MODAL,
+              newValue: true,
+              newProps: {
+                event: incentiveEvent,
+              },
+            })
+          );
```

Comment:
For first rule creation event, if it is not redirect or response then only it will trigger the first rule completed modal, else it will trigger the individual modal